lib/broadway.ex

```diff
@@ -710,7 +710,7 @@ defmodule Broadway do
   Returned by `start_link/2`.
   """
   @type on_start() :: {:ok, pid()} | :ignore | {:error, {:already_started, pid()} | term()}
-  @type name :: atom() | {:via, module(), term()}
+  @type name :: atom() | pid() | {:via, module(), term()}
```

Comment:
I don't believe this change is quite right. We now have two names: one used for registration, which can be pids, and one used for access, which cannot be pid. Maybe introduce a:

    @type server :: name() | pid()

Rename is_broadway_name to is_broadway_server and use Broadway.server() where appropriate?